### PR TITLE
adds zsh completion to iocage

### PR DESCRIPTION
**Pull request by wxcafe**
- [x] Explain the feature
    Adds ZSH completion, as explained in #160. I made a new dir for the zsh completion, and made setup.py copy it to the right place if it exists.
- [x] Read [CONTRIBUTING.md](https://github.com/iocage/iocage/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by skarekrow** [comment]
Awesome! Thank you

File: zsh-completion/_iocage [hunk 1]
```diff
@@ -0,0 +1,324 @@
+#compdef iocage
+
+## Written by Clément 'wxcafe' Hertling <clement.hertling@wxcafe.net>
+## on Oct-15-2017. Feel free to improve on this
+
+_iocage_list_jails() {
+  local -a jails
+  jails=($(iocage list -H | cut -f 2))
```
Comment (igalic) [inline zsh-completion/_iocage:8]: we can probably do this with `-o`; so no need to `cut`

File: zsh-completion/_iocage [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+}
+
+_iocage_list_releases() {
+  local -a releases
+  releases=($(iocage list -H -b))
+  _describe 'release' releases
+}
+_iocage_list_templates() {
+  local -a templates
+  templates=($(iocage list -H -t | cut -f 2))
```
Comment (igalic) [inline zsh-completion/_iocage:19]: again, use `-o` instead of `cut`

File: zsh-completion/_iocage [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+_iocage_list_potential_releases(){
+  local -a releases
+  releases=($(iocage list -R | grep '^\[' | cut -d ' ' -f 2))
+  _describe 'release' releases
+}
+
+_iocage_list_jails_or_releases() {
+  local -a releases jails
+  releases=($(iocage list -H -b))
+  jails=($(iocage list -H | cut -f 2))
```
Comment (igalic) [inline zsh-completion/_iocage:32]: again, use `-o` instead of `cut`

File: zsh-completion/_iocage [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+  _describe 'jail_or_release' jails -- releases
+}
+
+_iocage_list_zpools() {
+  zpool list -H | cut -f 1 | while read zpool; do
```
Comment (igalic) [inline zsh-completion/_iocage:37]: again, use `-o` instead of `cut`

**Review by igalic — COMMENTED** [review]
does iocage not have `-o`?

**Comment by wxcafe** [comment]
well, it's neither in the man page nor in `iocage list --help`, and 

```
$ iocage list -H -o
Error: no such option: -o
$ zpool list -H -o
missing argument for 'o' option
```

**Pull request merged**